app/controllers/Module_marketplace.php

```diff
@@ -0,0 +1,425 @@
+<?php
+
+namespace munkireport\controller;
+
+use \Controller, \View, \Model;
+use munkireport\models\Module_marketplace_model;
+use Symfony\Component\Yaml\Yaml;
+use munkireport\models\Cache;
+
+class Module_marketplace extends Controller
+{
+    private $moduleMarketplace;
+
+    public function __construct()
+    {
+        $this->moduleMarketplace = getMrModuleObj();
```

Comment:
// Check authorization
        $this->authorized() || jsonError('Authenticate first', 403);
        $this->authorized('global') || jsonError('You need to be admin', 403);

Reply:
Added